src/promptflow/promptflow/exceptions.py

```diff
@@ -52,6 +52,7 @@ def __init__(
         self._message_format = message_format
         self._kwargs = kwargs
         super().__init__(self._message)
+        self._try_get_message()
```

Comment:
We may directly update self._message if message_format is provided but message is not provided, then return it when `self.message` is called.
So we can know whether the rendering logic is OK when initialization, and reduce the cost when accessing `self.message`

Reply:
Updated.